libs/api/domains/occupational-licenses/src/lib/occupationalLicenses.service.ts

```diff
@@ -26,149 +70,200 @@ export class OccupationalLicensesService {
   async getHealthDirectorateLicenseById(
     user: User,
     id: string,
-  ): Promise<HealthDirectorateLicense | null | undefined> {
+  ): Promise<OccupationalLicenseResponse> {
     try {
-      const licenses =
-        (await this.healthDirectorateApi.getHealthDirectorateLicense(user)) ??
-        []
+      const licenses = await this.healthDirectorateApi
+        .getHealthDirectorateLicense(user)
+        .catch((e) => {
+          if (e.status === 404) {
+            return []
+          }
+          throw new Error('Error getting health directorate license by id')
+        })
+        .then((licenses) => licenses ?? [])
```

Comment:
It is because the API responds with 404 error if there does not exists a license for that user. I do not want to log an error, when there is no license.